pkg/frontend/admin_supportedvmsizes_list_test.go

```diff
@@ -0,0 +1,86 @@
+package frontend
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"testing"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/api/validate"
+	"github.com/Azure/ARO-RP/pkg/metrics/noop"
+)
+
+func TestSupportedvmsizes(t *testing.T) {
+	ctx := context.Background()
+	mastervmsizes := validate.SupportedVMSizesByRole(validate.VMRoleMaster)
+	workervmsizes := validate.SupportedVMSizesByRole(validate.VMRoleWorker)
+
+	type test struct {
+		name           string
+		vmRole         string
+		wantStatusCode int
+		wantResponse   *map[api.VMSize]api.VMSizeStruct
+		wantError      string
+	}
+
+	for _, tt := range []*test{
+		{
+			name:           "vmRole is invalid",
+			vmRole:         "invalidVMRole",
+			wantStatusCode: http.StatusBadRequest,
+			wantError:      `400: InvalidParameter: : The provided vmRole 'invalidVMRole' is invalid. vmRole can only be master or worker`,
+			wantResponse:   nil,
+		},
+		{
+			name:           "vmRole is empty",
+			vmRole:         "",
+			wantStatusCode: http.StatusBadRequest,
+			wantError:      `400: InvalidParameter: : The provided vmRole '' is invalid. vmRole can only be master or worker`,
+			wantResponse:   nil,
+		},
+		{
+			name:           "master as vmRole",
+			vmRole:         "master",
+			wantStatusCode: http.StatusOK,
+			wantError:      "",
+			wantResponse:   &mastervmsizes,
+		},
+		{
+			name:           "worker as vmRole",
+			vmRole:         "worker",
+			wantStatusCode: http.StatusOK,
+			wantError:      "",
+			wantResponse:   &workervmsizes,
+		},
+	} {
+		t.Run(tt.name, func(t *testing.T) {
+			ti := newTestInfra(t).WithSubscriptions().WithOpenShiftClusters()
+			defer ti.done()
+			f, err := NewFrontend(ctx, ti.audit, ti.log, ti.env, nil, nil, nil, nil, nil, api.APIs, &noop.Noop{}, nil, nil, nil, nil, nil)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			go f.Run(ctx, nil, nil)
+			resp, b, err := ti.request(http.MethodGet,
+				fmt.Sprintf("https://server/admin/supportedvmsizes?vmRole=%s", tt.vmRole),
+				nil, nil)
```

Comment:
you don't need the testinfra, you can just create an empty `frontend` struct and just test the function directly. You can just call `frontend.GetVMSize` or whatever that function is called instead of creating a whole http request.

Reply:
Made the changes.